src/main/java/org/apache/maven/plugins/clean/CleanMojo.java

```diff
@@ -229,8 +248,16 @@ else if ( fast && multiModuleProjectDirectory != null )
                         + "this plugin, or the 'maven.clean.fastDir' user property to be set." );
             }
         }
+        if ( fast && !FAST_MODE_BACKGROUND.equals( fastMode )
+                  && !FAST_MODE_AT_END.equals( fastMode )
+                  && !FAST_MODE_DEFER.equals( fastMode ) )
+        {
+            getLog().warn( "Illegal value '" + fastMode + "' for fastMode. Allowed values are '"
+                    + FAST_MODE_BACKGROUND + "', '" + FAST_MODE_AT_END + "' and '" + FAST_MODE_DEFER + "'. "
+                    + "Reverting to use the default value '" + FAST_MODE_BACKGROUND + "'." );
+        }
```

Comment:
Shouldn't this rather fail with an exception?

Reply:
I was hesitating... so I don't have a strong opinion on that. I've changed to throw an exception.